### PR TITLE
plugin-api v1

### PR DESCRIPTION
**Pull request by utkobe**
This is to initialize the plugin-api package

@wordcount feel free to check packages/plugin-api/src/model.ts to check how to use boa

let's work together in this branch on plugin apis

### CODE REVIEW

File: packages/plugin-api/src/model.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const boa = require('@pipcook/boa');
+import * as fs from 'fs-extra';
+import * as path from 'path'
+
+export function gray(src: string, dest: string) {
+  fs.ensureDirSync(dest);
```
Comment (yorkie) [inline packages/plugin-api/src/model.ts:6]: Change to async function?

File: packages/plugin-api/src/model.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  const cv2 = boa.import('cv2');
```
Comment (yorkie) [inline packages/plugin-api/src/model.ts:7]: Moving to file scope?

File: packages/plugin-api/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "dist"
+  ],
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "npm run clean && npm run compile",
+    "clean": "rm -rf ./dist && rm -rf tsconfig.tsbuildinfo",
+    "compile": "tsc -b tsconfig.json"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (yorkie) [inline packages/plugin-api/package.json:17]: Missing the license and author.

**Comment by yorkie** [comment]
@utkobe May I ask you to rebase your commits, it seems there are some comments belong to #56, which should not be in another PR.

**Comment by yorkie** [comment]
There are some errors from model.ts, @wordcount please have a look:

```
/Users/runner/runners/2.168.0/work/pipcook/pipcook/packages/plugin-api/src/model.ts
##[error]   6:7   error  A space is required after '{'                  object-curly-spacing
##[error]   6:21  error  A space is required before '}'                 object-curly-spacing
##[error]   6:52  error  Missing semicolon                              semi
##[error]  16:69  error  A space is required after ','                  comma-spacing
##[error]  25:2   error  Newline required at end of file but not found  eol-last
```

File: packages/plugin-api/src/model.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+const boa = require('@pipcook/boa');
```
Comment (yorkie) [inline packages/plugin-api/src/model.ts:4]:
```suggestion
import * as boa from '@pipcook/boa';
```

File: packages/cli/lib/init.js [hunk 5]
```diff
@@ -27,7 +27,9 @@ const init = async (cmdObj) => {
     if (commandExistsSync('tnpm')) {
       clientChoices.push('tnpm');
     }
-    if (clientChoices.length > 0) {
+    if (clientChoices.length === 1) {
+      client = clientChoices[0];
+    } else if (clientChoices.length > 1) {
```
Comment (yorkie) [inline packages/cli/lib/init.js:32]: This change is not necessary to be here.

File: packages/core/.gitignore [hunk 6]
```diff
@@ -1 +0,0 @@
-/node_modules
```
Comment (yorkie) [inline packages/core/.gitignore:1]: Why this change?

File: packages/plugin-api/package.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@pipcook/plugin-api",
+  "version": "0.5.0",
+  "description": "",
```
Comment (yorkie) [inline packages/plugin-api/package.json:4]: Add a description

File: packages/plugin-api/src/model.ts (hunk 4 above)
Comment (utkobe) [reply packages/plugin-api/src/model.ts:4]: Does boa support this format of import now. Probably need .d.ts files for typescript

**Comment by yorkie** [comment]
https://github.com/alibaba/pipcook/blob/master/package.json#L23 This needs to be updated.

File: packages/plugin-api/src/model.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+const cv2 = boa.import('cv2');
+const { word_tokenize } = boa.import('nltk.tokenize');
+const nltk = boa.import('nltk');
+
+
```
Comment (Txiaozhe) [inline packages/plugin-api/src/model.ts:9]: Redundant blank lines

File: packages/plugin-api/src/model.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+export function gray(src: string, dest: string) {
+  fs.ensureDirSync(dest);
+  const image = cv2.imread(src);
+  const gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY);
+  cv2.imwrite(path.join(dest, path.basename(src)), gray);
+}
+
+export function resize(src: string, dest: string, dim: [number, number]) {
```
Comment (Txiaozhe) [inline packages/plugin-api/src/model.ts:17]: [ number, number ] ??

**Comment by utkobe** [comment]
> https://github.com/alibaba/pipcook/blob/master/package.json#L23 This needs to be updated.

The tests would fail since the python libraries required by plugin-api are not installed by default

**Comment by yorkie** [comment]
> > https://github.com/alibaba/pipcook/blob/master/package.json#L23 This needs to be updated.
> 
> The tests would fail since the python libraries required by plugin-api are not installed by default

Could we introduce the "bip" tool to install them?

**Comment by utkobe** [comment]
> > > https://github.com/alibaba/pipcook/blob/master/package.json#L23 This needs to be updated.
> > 
> > 
> > The tests would fail since the python libraries required by plugin-api are not installed by default
> 
> Could we introduce the "bip" tool to install them?

I'm doing this in PR #100 . Do we do them separately or merge two PR?

**Comment by yorkie** [comment]
> I'm doing this in PR #100 . Do we do them separately or merge two PR?

Independent PR seems good to me, could you please add tests here once #100 gets merged?

**Comment by utkobe** [comment]
Sure :)

**Comment by yorkie** [comment]
We will move these works to #150, so closing this now.
